enhancements/insights/on-demand-data-gathering.md

```diff
@@ -0,0 +1,340 @@
+
+## Summary
+
+The Insights operator periodically gathers cluster data and sends it to Red Hat. The data is analyzed and the corresponding Insights analysis result is provided to the cluster. The OpenShift user can then see the number of active recommendations and go to the Insights advisor for
+more information. Right now, there is no way for a user to run Insights data gathering on demand - e.g when some of the Insights recommendations have been resolved.  
+
+## Motivation
+
+The motivation is to allow users (cluster admins) to run Insights data gathering on demand and thus enable to refresh the last Insights status in the cluster. 
+
+### User Stories
+- As an OCP user/cluster-admin I would like to trigger on-demand gathering so that I can refresh the current Insights status of my cluster and I don't have to wait 2 hours or delete the Insights operator pod. (e.g there were some active Insights health checks in the cluster, which should be fixed and I want to refresh the Insights status)
+- As an ACM (Advanced Cluster Manager) user/admin I would like to externally trigger on-demand gathering so that I can refresh the Insight status of my cluster (or multiple clusters I administrate)
+
+### Goals
+
+The goal is to provide OpenShift users/admins an easy way to run and trigger Insights data gathering on demand and refresh the Insights status of their cluster/s. 
+The way of triggering the new data gathering should be simple enough to e.g create a new button in the ACM (advanced cluster manager) refreshing the Insights status of the given cluster. 
+
+### Non-Goals
+
+- Change data gather frequency.
+- Change gathered data. 
+- Change operator status reporting (including metrics). This should work as previously.
+
+## Proposal
+
+The enhancement proposal introduces a new Insights `CustomResourceDefinition` type in the OpenShift API that will serve as the definition and trigger to new data gathering on demand. 
+This new CRD will build on and extend the existing `GatherConfig` (see the [Insights config API enhancement](../insights/insights-config-api.md)). The idea is that with each new CR instance of this resource, new data gathering will happen. 
+
+The proposal for the new `datagather.insights.openshift.io` resource is:
+
+```go
+type DataGather struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	Spec DataGatherSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status DataGatherStatus `json:"status"`
+}
+
+type DataGatherSpec struct {
+	// dataPolicy allows user to enable additional global obfuscation of the IP addresses and base domain
+	// in the Insights archive data. Valid values are "None" and "ObfuscateNetworking".
+	// When set to None the data is not obfuscated.
```

Comment:
```suggestion
	// When set to ClearText the data is not obfuscated.
```

?  Can be argued out on the API PR

Reply:
Yes good point with the matching. I agree we can update it, but is it part of this enhancement? I think it would make sense to update both API resources in separate PR or WDYT?

Reply:
I would update this one to be what we want, and make an issue on your Jira to update the old one, may as well get this one right first time right?

Reply:
OK

Reply:
Updated.